### DISCUSSION TITLE
Vulkan vs DX12 FPS

### DISCUSSION THREAD

**Discussion by SetheenDev** [gfx-rs/wgpu#2444] [Q&A] [answered]
Hello!  I'm referencing the tutorials from [learn-wgpu](https://sotrh.github.io/learn-wgpu/) and tweaking this basic [texture example](https://github.com/sotrh/learn-wgpu/tree/master/code/beginner/tutorial5-textures).  I see ~600 FPS using Vulkan and ~1600 using DX12.  I'm curious why:
- Performance difference between them is so stark
- Performance overall is low

[This OpenGL example](https://youtu.be/Yw7RpvmqvSE?t=315) from 2 years ago runs an entire game simulation with multiple textures at well over 2k FPS.  Removing the `draw_indexed` to just perform a clear color pass is about the same speed.  Is this expected?  Any flags or tweaks?  Maybe there's a better example I could profile?

Scenario:
- Compiled in release mode
- Default 800x600 windowed
- PresentMode - Immediate
- Windows 11
- Laptop NVidia RTX 3080
- 11th Gen Core i9

**Comment by cwfitzgerald** [comment, marked as answer]
This is quite common to see between GL and other apis. Vulkan in particular I've always noticed just runs kinda slow when doing nothing. This is very likely because of some component of the VK => Windows layer that nvidia provides just being a bit slower than going directly through DXGI or through the GL => Windows layer.

You'll very quickly find that this is by no means indicative of performance as it scales up.

**Reply by kvark** [reply]
In other words, Vulkan wasn't invented for doing faster screen clears. The more complex API interaction you do, the more Vulkan can help.

**Reply by SetheenDev** [reply]
10-4.  Appreciate your insight.  That makes sense.  Similar to something like deferred rendering, there's a lower baseline out of the gate but increased throughput when the action begins.  Thank you both
